app/controllers/users_controller.rb

```diff
@@ -21,6 +29,29 @@ def create
     end
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+    return if @user
+    flash[:warning] = t :show
+    redirect_to root_path
+  end
+
+  def update
+    @user = User.find_by id: params[:id]
+    if @user&.update_attributes user_params
+      flash[:success] = t :update
+      redirect_to @user
+    else
+      render :edit
+    end
+  end
+
+  def destroy
+    @user.destroy
+    flash[:success] = t :delete
+    redirect_to users_url
+  end
+
```

Comment:
Now you can see the 4 actions show, edit, update and destroy has the following line duplicated.
`@user = User.find_by id: params[:id]`
Can you try using **before_action** filter to remove this duplication